scripts/test/SANS/gui_logic/add_runs_presenter_test.py

```diff
@@ -0,0 +1,289 @@
+        generated_name = self._retrieve_generated_name_for(['1', '2', '3'])
+        self.assertEqual('3', generated_name)
+
+    def test_regenerates_correct_base_name_after_highest_removed(self):
+        run_summation = mock.Mock()
+        presenter = self._make_presenter(run_summation)
+        self._update_selection_model(self._make_fake_runs(['4', '5', '6']))
+        self._update_selection_model(self._make_fake_runs(['4', '5']))
+        self.view.sum.emit()
+        self.assertEqual('5', self._base_file_name_arg(run_summation))
+
+    def test_correct_base_name_after_set_by_user(self):
+        user_out_file_name = 'Output'
+        run_summation = mock.Mock()
+        presenter = self._make_presenter(run_summation)
+        self._update_selection_model(self._make_fake_runs(['4', '5', '6']))
+
+        self.view.out_file_name.return_value = user_out_file_name
+        self.view.outFileChanged.emit()
+        self.view.sum.emit()
+        self.assertEqual(user_out_file_name,
+                         self._base_file_name_arg(run_summation))
+
+    def test_base_name_not_reset_after_set_by_user(self):
+        run_summation = mock.Mock()
+        presenter = self._make_presenter(run_summation)
+        self._update_selection_model(self._make_fake_runs(['4', '5', '6']))
+
+        user_out_file_name = 'Output'
+        self.view.out_file_name.return_value = user_out_file_name
+        self.view.outFileChanged.emit()
+
+        self._update_selection_model(self._make_fake_runs(['4', '5']))
+        self.view.sum.emit()
+        self.assertEqual(user_out_file_name,
+                         self._base_file_name_arg(run_summation))
+
+    def test_sets_name_in_view_after_selection_update(self):
+        run_summation = mock.Mock()
+        presenter = self._make_presenter(run_summation)
+        self._update_selection_model(self._make_fake_runs(['4', '6', '5']))
+        self.view.set_out_file_name.assert_called_with('6')
+        self._update_selection_model(self._make_fake_runs(['5', '4']))
+        self.view.set_out_file_name.assert_called_with('5')
+
+
+class SumButtonTest(SelectionMockingTestCase):
+    def setUp(self):
+        self.setUpMockChildPresentersWithDefaultSummationSettings()
+        self.run_summation = self._make_mock_run_summation()
+        self.view = self._make_mock_view()
+        self.presenter = self._make_presenter()
+
+    def _make_presenter(self):
+        return AddRunsPagePresenter(
+            self._make_mock_run_summation(),
+            self._capture_on_change_callback(self.run_selector_presenter),
+            self._just_use_summation_settings_presenter(),
+            self.view,
+            None)
+
+    def test_enables_sum_button_when_row_added(self):
+        fake_run_selection = self._make_fake_runs(['5'])
+        fake_run_selection.has_any_runs.return_value = True
+        self._update_selection_model(fake_run_selection)
+        assert_called(self.view.enable_sum)
```

Comment:
I think it is better to use: `assert(self.view.enable_sum.call_count == 1)`

or what ever comparison you want.

Reply:
I also like to check that it is called with the right arguments but this may just be a personal preference thing.